src/ast/passes/semantic_analyser.cpp

```diff
@@ -2492,6 +2553,13 @@ void SemanticAnalyser::visit(AssignMapStatement &assignment)
   const std::string &map_ident = assignment.map->ident;
   auto type = assignment.expr->type;
 
+  if (assignment.map->wildcards)
+  {
+    LOG(ERROR, assignment.loc, err_)
+        << "Wildcards are only valid for delete() calls";
```

Comment:
Nit: perhaps a more maintainable error message is to say soemthing like "wildcards are not valid for assignments". So that if map wildcards gain more use cases, we don't need to update this messsage.